static/js/util/util_test.js

```diff
@@ -902,4 +903,30 @@ describe("utility functions", () => {
       }, 20)
     }, 20)
   })
+
+  describe("findObjByName", () => {
+    const obj = {
+      foo:  "bar",
+      bool: {
+        must: [
+          {
+            foo1: "baz"
+          },
+          {
+            foo2: "gaz"
+          }
+        ]
+      }
+    }
+
+    it("it deep search", () => {
+      const object = findObjByName(obj, "foo2")
+      assert.deepEqual(object, [{ foo2: "gaz" }])
+    })
+
+    it("it returns empty array when no match found", () => {
+      const object = findObjByName(obj, "foo3")
+      assert.equal(object.length, 0)
+    })
```

Comment:
it would be good to test that it also returns multiple matches if they are present, and test that it returns an object at the top level as well.

Reply:
(i.e. test that the object doesn't have to be deeply nested)